File: backend/src/main/java/com/pawconnect/backend/event/dto/EventResponse.java
package com.pawconnect.backend.event.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class EventResponse {
    private Long id;
    private String title;
    private String description;
    private LocalDateTime eventDateTime;
    private double latitude;
    private double longitude;
    private Long hostId;
    private List<Long> participantIds;
}

File: mobile/lib/src/models/event_response.dart
class EventResponse {
  final int id;
  final String title;
  final String? description;
  final DateTime eventDateTime;
  final double latitude;
  final double longitude;
  final int hostId;
  final List<int> participantIds;

  EventResponse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        title = json['title'],
        description = json['description'],
        eventDateTime = DateTime.parse(json['eventDateTime']),
        latitude = (json['latitude'] as num).toDouble(),
        longitude = (json['longitude'] as num).toDouble(),
        hostId = json['hostId'],
        participantIds = List<int>.from(json['participantIds'] ?? []);
}
